react/GalleryLayoutOption.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { ComponentType } from 'react'
+
+import { useCssHandles } from 'vtex.css-handles'
+import {
+  useSearchPageState,
+  useSearchPageStateDispatch,
+} from 'vtex.search-page-context/SearchPageContext'
+import { SWITCH_GALLERY_LAYOUT_TYPE } from './constants'
+
+interface OptionSlot {
+  isActive?: boolean
+}
+
+interface GalleryLayoutOptionProps {
+  Option: ComponentType<OptionSlot>
+  name: string
+}
+
+const CSS_HANDLES = ['galleryLayoutOption'] as const
+
+const GalleryLayoutOption: React.FC<GalleryLayoutOptionProps> = ({
+  Option,
+  name,
+}) => {
+  const handles = useCssHandles(CSS_HANDLES)
+  const { galleryLayout } = useSearchPageState()
+  const dispatch = useSearchPageStateDispatch()
+
+  const isActive = name === galleryLayout
+
+  const handleOptionClick = () => {
+    dispatch({
+      type: SWITCH_GALLERY_LAYOUT_TYPE,
+      args: { galleryLayout: name },
+    })
+  }
+
+  return (
+    <button
+      className={`${handles.galleryLayoutOption} grow dib br-100 pa2 mr2 ml2 bw0 pointer outline-0 bg-transparent`}
+      onKeyDown={() => handleOptionClick()}
+      onClick={() => handleOptionClick()}
+      aria-controls={'layout-option-items'}
+      aria-label={`Option for ${name} layout`}
+      data-testid="galleryLayoutOption"
```

Comment:
Do you really need this?

Reply:
Same question as @Klynger 

`data-testid` should be the last resort when testing, can we use the text inside of it to test?

Reply:
I don't think I do, since we don't test it yet. I based myself on a `slider-layout` component, so I just changed the value.